apps/cloudbroker/cloudbrokerlib/netmgr.py

```diff
@@ -337,12 +337,31 @@ def fw_list(self, gid, domain=None, **kwargs):
             result.append(vfwdict)
         return result
 
-    def fw_start(self, fwid, **kwargs):
+    def fw_start(self, fwid, resettype='restore', targetNid=None, **kwargs):
         """
         param:fwid firewall id
         param:gid grid id
         """
+        if resettype not in ['factory', 'restore']:
+            raise exceptions.BadRequest("Invalid value {} for resettype".format(resettype))
         fwobj = self._getVFWObject(fwid)
+        self.fw_delete(fwid, fwid.gid)
```

Comment:
we shouldnt do a delete if vm is currently in runnign state and vm is on targetNid or targetNid = none just return true then
this delete probably needs to be moved by to reset